### ISSUE TITLE
Frequent Error - Transport Error (Status Unknown)

### ISSUE THREAD

**Issue by piyush-bansal1989** [aerospike/php-client#41]
Hi Team

I'm writing to seek your assistance with a frequent error I'm encountering:

`status: Unknown, message: "transport error", details: [], metadata: MetadataMap { headers: {} }`

This error occurs frequently, and I'm unsure what might be missing to resolve it.
Could you please provide some insights on what this error message might indicate and any troubleshooting steps I can take to address it?

**Comment by piyush-bansal1989** [comment]
> Where do you encounter this error? As an exception in the PHP client? Do you have the result code to that exception? It seems to me that you are hitting a timeout error in part of the code chain, quite possibly in the GRPC layer, but without knowing more about your operating environment (the API you use, your policies and timeouts, the network quality and latency, etc) I cannot help you more.

Yes, as exception in php client, No result code throw in this case. 
Operating in aws ec2 instance, where 3 nodes of cluster is setup, and connecting through one client right now.
Currently I'm setting the following policies at put and get 

write policy
`$wp = new WritePolicy();
$wp->setRecordExistsAction(RecordExistsAction::Update());
      $wp->setMaxRetries(3);
      $wp->setSocketTimeout(5000);
      $wp->setSendKey(true);`

read policy
`        $rp = new ReadPolicy();
        $rp->setMaxRetries(3);
        $rp->setSocketTimeout(5000);
        $rp->setSendKey(true);`

**Comment by piyush-bansal1989** [comment]
> > Where do you encounter this error? As an exception in the PHP client? Do you have the result code to that exception? It seems to me that you are hitting a timeout error in part of the code chain, quite possibly in the GRPC layer, but without knowing more about your operating environment (the API you use, your policies and timeouts, the network quality and latency, etc) I cannot help you more.
> 
> Yes, as exception in php client, No result code throw in this case. Operating in aws ec2 instance, where 3 nodes of cluster is setup, and connecting through one client right now. Currently I'm setting the following policies at put and get
> 
> write policy `$wp = new WritePolicy(); $wp->setRecordExistsAction(RecordExistsAction::Update()); $wp->setMaxRetries(3); $wp->setSocketTimeout(5000); $wp->setSendKey(true);`
> 
> read policy ` $rp = new ReadPolicy(); $rp->setMaxRetries(3); $rp->setSocketTimeout(5000); $rp->setSendKey(true);`

@khaf : Can you please confirm about this? or will you suggest the configuration?

**Comment by khaf** [comment]
My colleague @vmsachin is working on this ticket. I'll let him respond.

**Comment by vmsachin** [comment]
@piyush-bansal1989 could you please tell me the exact setup of how you have setup the client and server.

**Comment by khaf** [comment]
@drZloy How did you remove the gRPC connection persistence?

**Comment by drZloy** [comment]
@khaf , please consider this commit in my fork: https://github.com/drZloy/aerospike-php-client/commit/07fc3b18ed9b3c28848f8c024c2171e857bd3fff
